### ISSUE TITLE
Kubernetes Ingress and sticky support

### ISSUE THREAD

**Issue by TerraTech** [containous/traefik#911]
Is it currently possible to configure _sticky_ support on a Kubernetes Ingress backend?

**Comment by errm** [comment]
Currently not, it sounds like we should make an option to switch sticky on with an annotation on the ingress...
We should wait until #386 merged to add this...

**Comment by TerraTech** [comment]
@errm Is #386 (UPXd Traefik) the correct issue?  They don't seem to be related.

**Comment by rrichardson** [comment]
expanding the annotations to be more flexible is definitely needed.  Currently the only annotations allowed are path and pathprefix :  https://github.com/containous/traefik/blob/master/provider/kubernetes.go#L138-L150 

I have a need to add a  HeaderRegexp  rule, it could be included in this change as well.

Unfortunately k8s ingress grammar offers little in the way of features, as a workaround I think it was suggested in another ticket (which I can't currently find) that we adopt some sort of scheme where we use a list of traefik key/val annotations,  like : 
```
traefik.frontend.rule.key: Host
traefik.frontend.rule.value: "foo"
traefik.frontend.rule.key: HeadersRegexp
traefik.frontend.rule.value: "Cookies, (foo|bar)"
traefik.backend.loadbalancer.key:sticky
traefik.backend.loadbalancer.value:true 
```

**Comment by rrichardson** [comment]
Also, I now notice that this is a duplicate of https://github.com/containous/traefik/issues/674

**Comment by TerraTech** [comment]
@rrichardson sigh - I could have sworn I searched first before posting.  :/

**Comment by emilevauge** [comment]
>We should wait until #386 merged to add this...

I think @errm wanted to say #836 ;)

@rrichardson, indeed, we may wait for new labels to be implemented #444.

**Comment by ivanjaros** [comment]
What does "sticky" mean?

**Comment by rrichardson** [comment]
For some given key (usually a session-id)  the load balancer will continue
to send every request that contains that key to the same upstream endpoint.


e.g.  you have 3 upstream services,  1, 2 and 3.

a request comes in with  a Cookie  of session_id=aaaabbbbcccc

if it hasn't seen  session_id=aaaabbbbcccc before, it will pick the next
upstream, round-robin style.   Let's say it picks '2' it then sends the
request there.

Another request comes in, it contains session_id=aaaabbbbcccc,  it will
send that request to '2'

Another request comes it,  it is session_id=eeeeffffgggg,  it hasn't seen
this session_id before, so it picks an upstream, round-robin style,  etc.
etc.

**Comment by ivanjaros** [comment]
Oh, you mean session affinity. 
Just out of curiosity, is this needed because each instance has its own database?

**Comment by rrichardson** [comment]
its own, local state of some sort, it makes it so it doesn't have to share
some of that state in a central database.

**Comment by errm** [comment]
Fixed in #1068
